etw/cs/traceloggingdynamic/TraceLoggingDynamic.cs

```diff
@@ -1803,56 +2046,69 @@ private static UInt64 DateTimeToFileTime(DateTime value)
             return valueTicks < 504911232000000000 ? 0u : (UInt64)valueTicks - 504911232000000000u;
         }
 
-        private void ResetEvent()
+        private unsafe void ResetEvent()
         {
             Debug.Assert(this.name.IndexOf('\0') < 0, "Event name must not have embedded NUL characters.");
-
             this.data.Reset();
             this.metadata.Reset();
 
+            int pos = 2; // skip UInt16 MetadataSize
+
+            var eventName = this.name;
+            var eventNameLength = eventName.Length;
+            var eventNameMaxByteCount = this.utf8NameEncoding.GetMaxByteCount(eventNameLength);
             if ((this.tag & 0x0FE00000) == tag)
             {
                 // Event tag fits in 7 bits.
-                this.ResetEventSkipTag(2 + 1);
-                this.metadata.data[2] = unchecked((byte)(this.tag >> 21));
+                this.metadata.ReserveSpaceFor(eventNameMaxByteCount + 2 + 1 + 1);

```

Comment:
Note: Inline ResetEventSkipTag into ResetEvent, and use unmanaged code to avoid redundant array bounds check. Small but noticeable performance improvement.